lib/cloudregister/registerutils.py

```diff
@@ -530,73 +530,78 @@ def setup_registry(registry_fqdn, username, password):
 
 
 # ----------------------------------------------------------------------------
-def get_registry_credentials():
+def get_registry_credentials(set_new):
     """Read the registry credentials file
        and return its content or an empty dict."""
     config_json = {}
-    failed = False
-    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
-        file_error = False
-        try:
-            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
-                config_json = json.load(cred_json)
-        except OSError as error:
-            logging.info(
-                'Unable to open %s: %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, error, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            file_error = True
-        except json.decoder.JSONDecodeError:
-            logging.info(
-                'Unable to parse existing %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            file_error = True
-
-        if file_error:
-            mv_file_cmd = 'mv -Z {} {}.bak'.format(
-                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-            ).split()
-            failed = exec_subprocess(mv_file_cmd)
-            message = 'File not preserved.' if failed else 'File preserved.'
-            logging.info(message)
+    failed = None
+    file_error = False
+    message = ''
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+            config_json = json.load(cred_json)
+    except OSError as error:
+        message = (
+            'Unable to open %s: %s, preserving file as %s.bak'
+            % (REGISTRY_CREDENTIALS_PATH, error, REGISTRY_CREDENTIALS_PATH)
+        )
```

Comment:
Something is not right here, either the merge didn't work or the diff is off, what is shown as the diff that we try to read the config file and if we cannot then we log a message that we could not preserve the file. Given such inconsistencies it is basically impossible to review the PR.

Reply:
I see
```python
    except OSError as error:
        message = (
            'Unable to open %s: %s, preserving file as %s.bak'
            % (REGISTRY_CREDENTIALS_PATH, error, REGISTRY_CREDENTIALS_PATH)
        )
```
which is part of the original intent
open the credentials file to update the credentials, 
opening failed, preserve the file <-- that is the part of that exception
create a new net credentials file with the info
  
 I can squash the commits into one, see if that help ?

Reply:
But the preserving of the file is not there, line 547 in the diff has a `)` from the log message, then line 548 has `if set_new`. There is no attempt to create `.bak`

Reply:
yes, there is ? if there was an [error](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/162/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R560), we do the preserve-file op

Reply:
Ok there has to be a way to handle this with reasonable code locality.

Reply:
we do the attempt to move the file when there are two exceptions, so we do it once instead of twice and log the message accordingly, we have used that style in our code base before